frontend/src/container/GridCardLayout/GridCard/WidgetGraphComponent.tsx

```diff
@@ -133,31 +129,42 @@ function WidgetGraphComponent({
 		[],
 	);
 
-	const onDeleteHandler = useCallback(() => {
-		const isEmptyWidget = widget?.id === 'empty' || isEmpty(widget);
-		const widgetId = isEmptyWidget ? layout[0].i : widget?.id;
+	const updateDashboardMutation = useUpdateDashboard();
+
+	const onDeleteHandler = (): void => {
+		const updatedWidgets = selectedDashboard?.data?.widgets?.filter(
+			(e) => e.id !== widget.id,
+		);
+
+		const updatedLayout =
+			selectedDashboard.data.layout?.filter((e) => e.i !== widget.id) || [];
+
+		const updatedSelectedDashboard: Dashboard = {
+			...selectedDashboard,
+			data: {
+				title: selectedDashboard.data.title,
+				description: selectedDashboard.data.description,
+				name: selectedDashboard.data.name,
+				tags: selectedDashboard.data.tags,
+				widgets: updatedWidgets,
+				layout: updatedLayout,
+				variables: selectedDashboard.data.variables,
+			},
```

Comment:
Can we do this instead of individually assigning each variable from data?
```suggestion
			data: {
				...selectedDashboard.data,
				widgets: updatedWidgets,
				layout: updatedLayout,
			},
```